checker/src/main/java/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -220,7 +220,7 @@ public Set<AnnotationMirror> getNarrowedAnnotations(
         Set<AnnotationMirror> result = AnnotationUtils.createAnnotationSet();
 
         if (narrowedTypeKind == TypeKind.CHAR) {
-            result.add(SIGNED);
```

Comment:
I cannot create a test case that fails on this line `result.add(SIGNED)`.
`getNarrowedAnnotations` is called in transfer function and in propagationtreeannotator. In transfer function `int x; char(x);` the `char(x)` is not considered as narrowing operation but a casting operation. So the transfer function `visitNarrowingConversion` in `CFAbstractTransfer.java` is not even called.

In `PropagationTreeAnnotator.java`, the `addAnnoOrBound` just throws the `SIGNED` annotation as it is not comparable to `UNSIGNED`.

Reply:
For example, `checker/tests/signedness/CharCast.java:6` reports an `assignment.type.incompatible` error.